frontend/app/services/IdentityService.scala

```diff
@@ -10,19 +10,85 @@ import monitoring.IdentityApiMetrics
 import play.api.Logger
 import play.api.Play.current
 import play.api.libs.json._
-import play.api.libs.ws.WS
+import play.api.libs.ws.{WS, WSResponse}
 import views.support.IdentityUser
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.concurrent.duration._
 import scala.util.{Failure, Success}
-import dispatch._, Defaults.timer
+import dispatch._
+import Defaults.timer
 
 case class IdentityServiceError(s: String) extends Throwable {
   override def getMessage: String = s
 }
 
+trait IdentityApi {
+
+  def getUserPasswordExists(headers:List[(String, String)], parameters: List[(String, String)]) : Future[Boolean] = {
+    val endpoint = "user/password-exists"
+    val url = s"${Config.idApiUrl}/$endpoint"
+    Timing.record(IdentityApiMetrics, "get-user-password-exists") {
+      WS.url(url).withHeaders(headers: _*).withQueryString(parameters: _*).withRequestTimeout(1000).get().map { response =>
+        recordAndLogResponse(response.status, "GET user-password-exists", endpoint)
+        (response.json \ "passwordExists").asOpt[Boolean].getOrElse(throw new IdentityApiError(s"$url did not return a boolean"))
+      }
+    }
+  }
+
+
+
+  def get(endpoint: String, headers:List[(String, String)], parameters: List[(String, String)]) : Future[Option[IdUser]] = {
+    Timing.record(IdentityApiMetrics, "get-user") {
+      val url = s"${Config.idApiUrl}/$endpoint"
+      WS.url(url).withHeaders(headers: _*).withQueryString(parameters: _*).withRequestTimeout(1000).get()
+        .recover { case e =>
+          Logger.error("Failure trying to retrieve user data", e)
+          throw e
+        }
+        .map { response =>
+          recordAndLogResponse(response.status, "GET user", endpoint)
+          val jsResult = (response.json \ "user").validate[IdUser]
+          if (jsResult.isError) Logger.error(s"Id Api response on $url : ${response.json.toString} json problems: $jsResult")
+          jsResult.asOpt
+        }
+        .recover { case e =>
+          Logger.error("Failure trying to deserialise user data", e)
+          None
+        }
+    }
+  }
+
+  def post(endpoint: String, data: Option[JsObject], headers: List[(String, String)], parameters: List[(String, String)], metricName: String): Future[Int] = {
+    Timing.record(IdentityApiMetrics, metricName) {
+      val requestHolder = WS.url(s"${Config.idApiUrl}/$endpoint").withHeaders(headers: _*).withQueryString(parameters: _*).withRequestTimeout(5000)
+      val response = requestHolder.post(data.getOrElse(JsNull))
+      response.foreach(r => recordAndLogResponse(r.status, s"POST $metricName", endpoint ))
+      response.map(_.status)
+        .andThen {
+          case Success(status) =>
+            if ((status / 100) != 2) // non 2xx code
```

Comment:
I know you've just moved this from elsewhere in the file, but we should be able to write this line so that it is simple enough to not need a comment.